File: packages/plugin-agent/src/server/tools/system.ts
import type { AgentToolDefinition, AgentToolHandler, AgentToolContext, AgentToolResult } from '@purplesector/plugin-api';

export const systemToolDefinitions: AgentToolDefinition[] = [
  {
    name: 'getCurrentContext',
    description: 'Get the current application context — what page the user is on, which event/session/lap is selected.',
    inputSchema: { type: 'object', properties: {} },
    category: 'system',
    mutating: false,
  },
];

export function createSystemToolHandlers(_prisma: any): Record<string, AgentToolHandler> {
  return {
    async getCurrentContext(_args: Record<string, unknown>, ctx: AgentToolContext): Promise<AgentToolResult> {
      return {
        success: true,
        data: ctx.appContext ?? { page: 'unknown' },
        message: ctx.appContext
          ? `User is on page "${ctx.appContext.page}".`
          : 'No application context available.',
      };
    },
  };
}

File: apps/web/apps/agent/tools/system.py
"""Agent tools for system/context queries."""

from __future__ import annotations

from . import ToolDefinition, ToolContext, ToolResult, register_tools

definitions = [
    ToolDefinition(
        name="getCurrentContext",
        description="Get the current application context — what page the user is on, which event/session/lap is selected.",
        input_schema={"type": "object", "properties": {}},
        category="system",
    ),
]


async def get_current_context(args: dict, ctx: ToolContext) -> ToolResult:
    return ToolResult(
        success=True,
        data=ctx.app_context or {"page": "unknown"},
        message=f'User is on page "{ctx.app_context["page"]}".' if ctx.app_context else "No application context available.",
    )


handlers = {
    "getCurrentContext": get_current_context,
}

register_tools(definitions, handlers)
